fair/tools/unifiedtools.py

```diff
@@ -196,6 +199,86 @@ def create_output_dataframes(inp_df, gas_np, RF_np, T_np, alpha_np, ext_forcing_
+
+    model_region_scenario_array = np.unique(inp_df.timeseries().index.droplevel(('unit','variable')).to_numpy())
+    if len(model_region_scenario_array) > 1:
+        raise Exception('Error: More than one Model, Region + Scenario combination input passed')
+    model_region_scenario = np.array(model_region_scenario_array[0])
+
+    time_index = inp_df.timeseries().columns
+
+    data_array = np.array([[]])
+    for i in range(len(gas_array)):
+        gas_name = gas_array[i]
+        
+        gas_output_variable = output_function + "|" + gas_name
+        RF_output_variable = "Effective Radiative Forcing|" + gas_name
+        alpha_output_variable = "Alpha|" + gas_name
+
+        gas_output_unit = units[gas_output_variable]
+        RF_output_unit = units[RF_output_variable]
+        alpha_output_unit = units[alpha_output_variable]
+
+        gas_output_array = np.append(model_region_scenario,[gas_output_variable, gas_output_unit])
+        RF_output_array = np.append(model_region_scenario,[RF_output_variable, RF_output_unit])
+        alpha_output_array = np.append(model_region_scenario,[alpha_output_variable, alpha_output_unit])
+
+        gas_output_array = np.append(gas_output_array,gas_np[i])
+        RF_output_array = np.append(RF_output_array,RF_np[i])
+        alpha_output_array = np.append(alpha_output_array,alpha_np[i])
+
+        data_array = np.append(data_array,[gas_output_array],axis=int(i==0))
+        data_array = np.append(data_array,[RF_output_array],axis=0)
+        data_array = np.append(data_array,[alpha_output_array],axis=0)
+
+    ext_RF_output_variable = "Effective Radiative Forcing|External Forcing"
+    ext_RF_output_unit = units[ext_RF_output_variable]
+    ext_RF_output_array = np.append(model_region_scenario,[ext_RF_output_variable, ext_RF_output_unit])
+    ext_RF_output_array = np.append(ext_RF_output_array,ext_forcing_np)
+    data_array = np.append(data_array,[ext_RF_output_array],axis=0)
+
+    total_RF_np = RF_np.sum(axis=0) + ext_forcing_np
+    total_RF_output_variable = "Effective Radiative Forcing"
+    total_RF_output_unit = units[total_RF_output_variable]
+    total_RF_output_array = np.append(model_region_scenario,[total_RF_output_variable, total_RF_output_unit])
+    total_RF_output_array = np.append(total_RF_output_array,total_RF_np)
+    data_array = np.append(data_array,[total_RF_output_array],axis=0)
+
+    T_output_variable = "Surface Temperature"
+    T_output_unit = units[T_output_variable]
+    T_output_array = np.append(model_region_scenario,[T_output_variable, T_output_unit])
+    T_output_array = np.append(T_output_array,T_np)
+    data_array = np.append(data_array,[T_output_array],axis=0)
+
+    output_df = pd.DataFrame(data_array, columns = pyam.IAMC_IDX + time_index.to_list(),)
+    
+
+
+    pyam_df = pyam.IamDataFrame(output_df)
+
+    output_df = pyam.IamDataFrame(pyam_df.append(inp_df))
```

Comment:
It'll be quicker to append using pandas than pyam i.e. you only want to call `pyam.IamDataFrame` once (rather than twice as it is now)